hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/AccountBalanceSigningUtils.java

```diff
@@ -0,0 +1,124 @@
+     * <p>
+     * Should be called before using stream utilities
+     * <p>
+     * Do <strong>NOT</strong> call this during standard node operation, as it will interfere with static settings
+     */
+    public static void initializeSystem() {
+        BootstrapUtils.setupConstructableRegistry();
+
+        // we don't want deserialization to fail based on any of these settings, not needed for services
+        //        SettingsCommon.maxTransactionCountPerEvent = Integer.MAX_VALUE;
+        //        SettingsCommon.maxTransactionBytesPerEvent = Integer.MAX_VALUE;
+        //        SettingsCommon.transactionMaxBytes = Integer.MAX_VALUE;
+        //        SettingsCommon.maxAddressSizeAllowed = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Generates a signature file for the account balance file
+     *
+     * @param signatureFileDestination the full path where the signature file will be generated
+     * @param streamFileToSign         the stream file to be signed
+     * @param keyPair                  the keyPair used for signing
+     * @return true if the signature file was generated successfully, false otherwise
+     */
+    public static boolean signAccountBalanceFile(
+            @NonNull final Path signatureFileDestination,
+            @NonNull final Path streamFileToSign,
+            @NonNull final KeyPair keyPair) {
+
+        Objects.requireNonNull(signatureFileDestination, "signatureFileDestination must not be null");
+        Objects.requireNonNull(streamFileToSign, "streamFileToSign must not be null");
+        Objects.requireNonNull(keyPair, "keyPair must not be null");
+
+        try {
+            final Hash entireHash = computeEntireHash(streamFileToSign.toFile());
+            final byte[] fileHashByte = entireHash.getValue();
+            final byte[] signature = sign(fileHashByte, keyPair);
+
+            generateSigBalanceFile(signatureFileDestination.toFile(), signature, fileHashByte);
+
+            System.out.println("Generated signature file: " + signatureFileDestination);
+
+            return true;
+        } catch (final SignatureException | IOException e) {
+            System.err.println("Failed to sign file " + streamFileToSign.getFileName() + ". Exception: " + e);
+            return false;
+        } catch (final InvalidKeyException | NoSuchAlgorithmException | NoSuchProviderException e) {
+            throw new RuntimeException("Irrecoverable error encountered", e);
+        }
+    }
+
+    private static void generateSigBalanceFile(final File filePath, final byte[] signature, final byte[] fileHash) {
+        try (final BufferedOutputStream output = new BufferedOutputStream(new FileOutputStream(filePath))) {
+            output.write(TYPE_FILE_HASH);
+            output.write(fileHash);
+            output.write(TYPE_SIGNATURE);
+            output.write(integerToBytes(signature.length));
+            output.write(signature);
+            output.flush();
```

Comment:
Not strictly necessary since `BufferedOutputStream` will flush when auto-closed.  (Actually, it's a superclass that'll do it.)

Reply:
yea, just to be sure here.